client/src/controllers/PreviewController.ts

```diff
@@ -371,7 +371,12 @@ export class PreviewController extends Controller<HTMLElement> {
   autoUpdateIntervalValueChanged() {
     // If the value is changed, only update the interval if it's currently active
     // as we don't want to start the interval when the panel is hidden
-    if (this.updateInterval) this.addInterval();
+    if (this.updateInterval) {
+      this.addInterval();
+    } else if (!this.autoUpdateIntervalValue) {
+      // If the auto-update interval is unset, clear the interval
+      this.clearInterval();
+    }
```

Comment:
This change around the auto update interval debouncing is unrelated to the headless support. It was just me ensuring that any change to the auto-update-interval-value that happens after the initialisation is respected, because I had nothing better to do 🤷